src/fides/api/ctl/migrations/versions/144d9b85c712_update_consent_mechanism.py

```diff
@@ -0,0 +1,114 @@
+"""update consent mechanism
+
+Revision ID: 144d9b85c712
+Revises: 8188ddbf8cae
+Create Date: 2023-04-12 14:57:07.532660
+
+"""
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "144d9b85c712"
+down_revision = "8188ddbf8cae"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    """
+    Extra steps added here to avoid the error that enums have to be committed before they can be used.  This
+    avoids having to commit in the middle of this migration and lets the entire thing be completed in a single transaction
+    """
+    op.execute("alter type consentmechanism rename to consentmechanismold")
+
+    op.execute(
```

Comment:
oh, this is clever! what i've [done in the past](https://github.com/ethyca/fides/blob/main/src/fides/api/ctl/migrations/versions/2f6aa5fe797a_update_messaging_service_type_enum.py#L29) is create a temporary column, which is similar but feels a bit clunkier than what you have here

Reply:
awesome, thanks for your review!